addons/sale/models/sale_order.py

```diff
@@ -1202,11 +1212,40 @@ def _send_order_notification_mail(self, mail_template):
             # sending mail in sudo was meant for it being sent from superuser
             self = self.with_user(SUPERUSER_ID)
 
-        self.with_context(force_send=True).message_post_with_source(
-            mail_template,
-            email_layout_xmlid='mail.mail_notification_layout_with_responsible_signature',
-            subtype_xmlid='mail.mt_comment',
-        )
+        async_send = str2bool(self.env['ir.config_parameter'].sudo().get_param('sale.async_emails'))
+        cron = self.env.ref('sale.send_pending_emails_cron', raise_if_not_found=False)
+        if async_send and cron and not self.env.context.get('sale_is_async_email', False):
+            # Schedule the email to be sent asynchronously.
+            self.pending_email_template_id = mail_template
+            if not cron.active:
+                cron.active = True  # The cron is created inactive to save resources.
```

Comment:
Feedback wanted: Should I do this in overrides of the CRUD methods of `ir.config.param` to also disable the cron when the param is removed?

Reply:
> Yes, you should

-- next day me